### ISSUE TITLE
WebPart not working after Asset

### ISSUE THREAD

**Issue by Grahamworth** [SharePoint/sp-dev-docs#1078]
#### Category
- [ ] Question
- [ ] Typo
- [ x] Bug
- [ ] Additional article idea

#### Expected or Desired Behavior
I create a webpart using Sp framework (react) which contains a list asset, when deploying to a site the asset is created and the webpart can be added to pages.

#### Observed Behavior
When deploying just the webpart (no asset) I can add the webpart to pages.
When adding the asset feature text to the package-solution, the asset is created but the webpart is no longer visible when trying to add to a page

#### Steps to Reproduce
Deploying: 
1. Create SP framework webpart (following ms guide)
2. Add Asset schema and element (flowing ms guide )
3. Add Feature to solution: 
`{
  "$schema": "https://dev.office.com/json-schemas/spfx-build/package-solution.schema.json",
  "solution": {
    "name": "portal-client-side-solution",
    "id": "085054e3-9898-44e4-b4eb-4fdce820c32c",
    "version": "1.0.1.13",
    "features": [{
      "title": "O365 Portal Page",
      "description": "asset-deployment-webpart-client-side-solution",
      "id": "523fe887-ced5-4036-b564-8dad5c6c6e24",
      "version": "1.0.0.0",
      "assets": {        
        "elementManifests": [
          "elements.xml"
        ],
        "elementFiles":[
          "schema.xml"
        ]
      }
    }]
  },
  "paths": {
    "zippedPackage": "solution/portal.sppkg"
  }
}
4. Gulp bundle --ship
5. Gulp package-solution --ship
6. deploy solution
7. add app to site
8. asset creates, unable to add webpart to page, webpart visable in webpart list
(if i remove the features section, the webpart is available after deploy)

Serve
1. gulp serve
2. access workbench on site
3. webpart visible to add ( with or without feature)

Any help in getting this working would be fantastic

**Comment by VesaJuvonen** [comment]
thx @Grahamworth for reporting this. Just to ensure that we can do quick repro on this, can you provide still following details - what was the site template you used for the testing? - Publishing, Team or group site? - also - were you using classic or modern pages or was this visible in classic pages - web part pages or publishing pages? - This seems to be related to the tenant scope settings and/or asset packaging, but it would help us to know the exact environment details for the repro steps. Thx.

**Comment by Grahamworth** [comment]
I used a Team Site, I could not see the webpart in classic webparts or modern pages. 

Let me know if you need anything else

Thank you

**Comment by VesaJuvonen** [comment]
@Grahamworth - any change if you can check if the feature is activated in the site features UI - if that's not being activated after the solution is installed, that indicates that the element.xml file is not valid or have some schema issues. If you can share the sppkg package, that would help on checking the issue as well.

**Comment by mcmynn83** [comment]
Can you take a look at the site features and ensure that your feature is actually activated.  My intuition is telling me that there is a bug in your feature xml and the feature failed to activate.  If that happens your webpart will not show up because the feature is not active.

**Comment by Grahamworth** [comment]
zip file: 
[portal.zip](https://github.com/SharePoint/sp-dev-docs/files/1538226/portal.zip)

Schema and elements xml below:
**schema.xml**
`<List xmlns:ows="Microsoft SharePoint" Title="Service Portal" EnableContentTypes="TRUE" FolderCreation="FALSE" Direction="$Resources:Direction;" Url="Lists/Service Portal" BaseType="0" xmlns="http://schemas.microsoft.com/sharepoint/">
  <MetaData>
    <ContentTypes>
      <ContentTypeRef ID="0x010042D0C1C200A14B6887742B6344675C8B" />
    </ContentTypes>
    <Fields></Fields>
    <Views>
      <View BaseViewID="1" Type="HTML" WebPartZoneID="Main" DisplayName="$Resources:core,objectiv_schema_mwsidcamlidC24;" DefaultView="TRUE" MobileView="TRUE" MobileDefaultView="TRUE" SetupPath="pages\viewpage.aspx" ImageUrl="/_layouts/images/generic.png" Url="AllItems.aspx">
        <XslLink Default="TRUE">main.xsl</XslLink>
        <JSLink>clienttemplates.js</JSLink>
        <RowLimit Paged="TRUE">30</RowLimit>
        <Toolbar Type="Standard" />
        <ViewFields>
          <FieldRef Name="LinkTitle"></FieldRef>
          <FieldRef Name="SP_Description"></FieldRef>
          <FieldRef Name="SP_Link"></FieldRef>
          <FieldRef Name="SP_Icon"></FieldRef>
          <FieldRef Name="SP_Order"></FieldRef>
          <FieldRef Name="SP_Information"></FieldRef>
        </ViewFields>
        <Query>
          <OrderBy>
            <FieldRef Name="ID" />
          </OrderBy>
        </Query>
      </View>
    </Views>
    <Forms>
      <Form Type="DisplayForm" Url="DispForm.aspx" SetupPath="pages\form.aspx" WebPartZoneID="Main" />
      <Form Type="EditForm" Url="EditForm.aspx" SetupPath="pages\form.aspx" WebPartZoneID="Main" />
      <Form Type="NewForm" Url="NewForm.aspx" SetupPath="pages\form.aspx" WebPartZoneID="Main" />
    </Forms>
  </MetaData>
</List>`

**elements.xml**
`<?xml version="1.0" encoding="utf-8"?>
<Elements xmlns="http://schemas.microsoft.com/sharepoint/">
    <Field ID="{060E50AC-E9C1-4D3C-B1F9-DE0BCAC300F6}"
            Name="SP_Description"
            DisplayName="Description"
            Type="Text"
            Required="FALSE"
            Group="SP_Columns" />
    <Field ID="{d4a5c692-d5a3-4f29-8173-e49a4515883a}"
            Name="SP_Link"
            DisplayName="Button Link"
            Type="URL"
            Required="FALSE"
            Group="SP_Columns" />
    <Field ID="{4fb88b29-7a0c-44d6-b7e2-7dbd39c2ec91}"
            Name="SP_Icon"
            DisplayName="Icon Link"
            Type="URL"
            Required="FALSE"
            Group="SP_Columns" />
    <Field ID="{68f67eff-1896-4486-acfb-79182f897c6d}"
            Name="SP_Order"
            DisplayName="Order"
            Type="Number"
            Decimals="0"
            Min="0"
            Required="FALSE"
            Group="SP_Columns" />
    <Field ID="{ca7b5c50-c7ad-444a-88d4-798490735155}"
            Name="SP_Information"
            DisplayName="Information Link"
            Type="URL"
            Required="FALSE"
            Group="SP_Columns" />
    <ContentType ID="0x010042D0C1C200A14B6887742B6344675C8B" 
            Name="Service Portal" 
            Group="Service Portal Content Types" 
            Description="Required for adding new list items">
        <FieldRefs>
            <FieldRef ID="{060E50AC-E9C1-4D3C-B1F9-DE0BCAC300F6}" />
            <FieldRef ID="{d4a5c692-d5a3-4f29-8173-e49a4515883a}" />
            <FieldRef ID="{4fb88b29-7a0c-44d6-b7e2-7dbd39c2ec91}" />
            <FieldRef ID="{68f67eff-1896-4486-acfb-79182f897c6d}" />
            <FieldRef ID="{ca7b5c50-c7ad-444a-88d4-798490735155}" />
        </FieldRefs>
    </ContentType> 
    <ListInstance 
            CustomSchema="schema.xml"
            FeatureId="00bfea71-de22-43b2-a848-c05709900100"
            Title="Service Portal" 
            Description="Stores Links for your Service Portal"
            TemplateType="100"
            Url="Lists/ServicePortal">
    </ListInstance>
</Elements>`

**Comment by mcmynn83** [comment]
@Grahamworth  - I have added your package to a test tenant and the Portal webpart is available to be added to a page.  I am not seeing this issue repro.  If you would like further help with this please capture a fiddler trace of the page loading so that we can see the request to get the client side web parts and see what is being returned.

**Comment by Grahamworth** [comment]
I am still getting the issue, struggling to figure out why.
To produce the below fiddler capture I....

1. Deployed the app with the feature asset not included
2. Added the webpart to the default page
3. Increment the version and Deployed the app with the feature asset included
4. enabled fidler and browsed to the default page (webpart now missing)

[output.zip](https://github.com/SharePoint/sp-dev-docs/files/1547927/output.zip)

**Comment by Grahamworth** [comment]
Oh I also get some errors I did not notice before: 
![image](https://user-images.githubusercontent.com/13215345/33833547-40a4337c-de80-11e7-9815-c37e0401e98e.png)

**Comment by mcmynn83** [comment]
@Grahamworth - We didn't notice this was an upgrade scenario.  It looks like you are missing the steps in order to actually add an upgrade action as well as upgrade the feature in the site collection so that the new items actually show up.  Please review the documentation at:
https://docs.microsoft.com/en-us/sharepoint/dev/spfx/web-parts/get-started/provision-sp-assets-from-package 

If after following these steps you still are seeing errors please open a new thread (as comments on this closed thread will not be seen on our systems).

**Comment by paigeflourin** [comment]
Hi,

I am also having issues not on the upgrade but on the installation of the app. the list /asset is being created but i cannot find the app in the webpart gallery when i try to add it.,

**Comment by Grahamworth** [comment]
I ended up creating a new webpart, copying my webpart code across, and re-writing the asset from scratch, at first just with one field to make sure it worked, then I added the rest of the fields, increment the feature version and deployed it as an update.

I'm not 100% sure what went wrong in my first deployment and why the webpart would not display, but I think thier must have been an error in my element or schema file. Either way, the behavior was very strange as it gave no error, it just didnt show the webpart. 

Anyway, one thing I learnt was to write my code to be easily transferable
